enhancements/windows-containers/windows-node-egress-proxy.md

```diff
@@ -0,0 +1,347 @@
+or by modifying certificates present in their [trustedCA ConfigMap](https://docs.openshift.com/container-platform/4.12/security/certificates/updating-ca-bundle.html#ca-bundle-replacing_updating-ca-bundle).
+
+In each case, Windows nodes can be joined to the cluster after altering proxy settings in which case proxy settings would
+be applied initial during node configuration. In the latter 2 scenarios, Windows nodes may already be existing on the
+cluster in which case they would need to be updated.
+
+### Risks, Drawbacks, and Mitigations
+
+The risks and mitigations are similar to those on the [Linux side of the cluster-wide proxy](https://github.com/openshift/enhancements/blob/master/enhancements/proxy/global-cluster-egress-proxy.md#risks-and-mitigations).
+Although cluster infra resources already do a best effort validation on the user-provided proxy URL schema and CAs,
+a user could provide non-functional proxy settings/certs. This would be propogated to their Windows nodes and workloads,
+taking down existing application connectivity and preventing new Windows nodes from being bootstrapped.
+
+The only drawbacks are the increased complexity of WMCO and the potential complexity of debugging customer cases that
+involve a proxy setup, since it would be extremely difficult to set up an accurate replicatation environment.
+This can be mitigated by proactively getting the development team, QE, and support folks familiar with the expected
+behavior of Windows nodes/workloads in proxied clusters, and comfortable spinning up their own proxied clusters.
+
+### API Extensions
+
+N/A, as no CRDs, admission and conversion webhooks, aggregated API servers, or finalizers will be added or modified.
+Only the WMCO will be extended which is an optional operator with its own lifecycle and SLO/SLAs, a 
+[tier 3 OpenShift API](https://docs.openshift.com/container-platform/4.12/rest_api/understanding-api-support-tiers.html#api-tiers_understanding-api-tiers).
+
+### Operational Aspects of API Extensions
+
+N/A
+
+#### Failure Modes & Support Procedures
+
+In general the support procedures for WMCO will remain the same. There are two underlying mechanisms we rely on, the
+publishing of proxy config to cluster resources and the comsuming of the published config. If either of the underlying
+mechanisms fail, Windows nodes will become proxy unaware. This could involve an issue with the user-provided proxy
+settings, the cluster network operator, OLM, or WMCO. This would result in all future egress traffic circumventing the
+proxy, which could affect inter-pod communication, existing application availablity, and security. Also, the pause image
+may not be able to be fetched, preventing new Windows nodes from being bootstrapped. This would require manual
+intervention from the cluster admin or a new release fixing whatever bug is causing the problem.
+
+## Design Details
+
```

Comment:
I see 'configuring the proxy settings in WMCO-managed components on Windows nodes (kubelet, containerd runtime)'  in the goals section, but I think we are missing it in the implementation. Can we add more detail here on what role WMCO plays in that?

Reply:
Basically there are 3 layers to the env vars: Machine, User, Process
- Env vars in the `User` layer inherit from `Machine` layer
- `Process` layer (e.g. Powershell.exe) inherits from both `Machine` and `User`
- Any subprocesses (containerd.exe etc) inherit from their parent process (powershell.exe)

So when we set the vars on the instance (`Machine` scope), the Windows services inherit them. I tried to explain with the `Windows services read environment from the system registry` line but I'll make this more explicit.

Reply:
I moved this stuff to the alternatives section and called it out there